### ISSUE TITLE
Warn about unimplemented methods in final class

### ISSUE THREAD

**Issue by ianliu** [microsoft/pyright#2131]
**Describe the bug**
A final class deriving an abstract class must implement all abstract methods, otherwise the class is "useless". Pyright should warn about unimplemented methods in final classes.

Mypy does warn about unimplemented abstract methods on final classes, as shown in this playground: https://mypy-play.net/?mypy=latest&python=3.10&gist=1dff236d04b7323f2ca95640d9afddc1

The error message is as follows:
```
main.py:9: error: Final class main.Bar has abstract attributes "foo"
Found 1 error in 1 file (checked 1 source file)
```

**Screenshots or Code**
```python
from typing import final
from abc import ABC, abstractmethod

class Foo(ABC):
    @abstractmethod
    def foo(self): pass

@final
class Bar(Foo): # foo method should be implemented!
    pass
```

**VS Code extension or command-line**
Running pyright from coc-pyright plugin for neovim. Pyright version is 1.1.158

**Comment by erictraut** [comment]
Thanks for the suggestion. I confirmed that mypy implements this check.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by ianliu** [comment]
Thanks @erictraut, appreciate the speed, this project is awesome!

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
